src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,330 @@
+```math
+p = (\gamma - 1) \left( \rho e - \frac{1}{2} \rho (v_1^2+v_2^2) \right)
+```
+as the pressure. The terms on the right hand side of the system above
+are built from the viscous stress tensor
+```math
+\underline{\tau} = \mu \left(\nabla\mathbf{v} + \left(\nabla\mathbf{v}\right)^T\right) - \frac{2}{3} \mu \left(\nabla\cdot\mathbf{v}\right)\underline{I}
+```
+where ``\underline{I}`` is the ``2\times 2`` identity matrix and the heat flux is
+```math
+\nabla q = -\kappa\nabla\left(T\right),\quad T = \frac{p}{R\rho}
+```
+where ``T`` is the temperature and ``\kappa`` is the thermal conductivity for Fick's law.
+Under the assumption that the gas has a constant Prandtl number
+the thermal conductivity is
+```math
+\kappa = \frac{\gamma \mu R}{(\gamma - 1)\textrm{Pr}}
+```
+
+In two spatial dimensions we require gradients for three quantities, e.g.,
+primitive quantities
+```math
+\nabla v_1,\, \nabla v_2,\, \nabla T
+```
+or the entropy variables
+```math
+\nabla w_2,\, \nabla w_3,\, \nabla w_4
+```
+where
+```math
+w_2 = \frac{\rho v_1}{p},\, w_3 = \frac{\rho v_2}{p},\, w_4 = -\frac{\rho}{p}
+```
+
+For this particular scaling the vicosity is set internally to be μ = 1/Re.
+Further, the nondimensionalization takes the density-temperature-sound speed as
+the principle quantities such that
+```
+rho_inf = 1.0
+T_ref = 1.0
+c_inf = 1.0
+p_inf = 1.0 / gamma
+u_inf = Ma_inf
+R = 1.0 / gamma
+```
+
+Other normalization strategies exist, see the reference below for details.
+- Marc Montagnac (2013)
+  Variable Normalization (nondimensionalization and scaling) for Navier-Stokes
+  equations: a practical guide
+  [CERFACS Technical report](https://www.cerfacs.fr/~montagna/TR-CFD-13-77.pdf)
+The scaling used herein is Section 4.5 of the reference.
+"""
+struct CompressibleNavierStokesEquations2D{RealT <: Real, E <: AbstractCompressibleEulerEquations{2}} <: AbstractCompressibleNavierStokesEquations{2, 4}
+  # TODO:
+  # 1) For now save gamma and inv(gamma-1) again, but could potentially reuse them from the Euler equations
+  # 2) Add NGRADS as a type parameter here and in AbstractEquationsParabolic, add `ngradients(...)` accessor function
+  gamma::RealT               # ratio of specific heats
+  inv_gamma_minus_one::RealT # = inv(gamma - 1); can be used to write slow divisions as fast multiplications
+  Re::RealT                  # Reynolds number
+  Pr::RealT                  # Prandtl number
+  Ma_inf::RealT              # free-stream Mach number
+  kappa::RealT               # thermal diffusivity for Fick's law
+
+  p_inf::RealT               # free-stream pressure
+  u_inf::RealT               # free-stream velocity
+  R::RealT                   # gas constant (depends on nondimensional scaling!)
+
+  equations_hyperbolic::E    # CompressibleEulerEquations2D
+end
+
+function CompressibleNavierStokesEquations2D(equations::CompressibleEulerEquations2D; Reynolds, Prandtl, Mach_freestream)
+  γ = equations.gamma
```

Comment:
```suggestion
  gamma = equations.gamma
```
To avoid non-ASCII characters?